lib/glimmer/tk/widget_proxy.rb

```diff
@@ -551,21 +562,91 @@ def handle_listener(listener_name, &listener)
       def on(listener_name, &listener)
         handle_listener(listener_name, &listener)
       end
-      
+
+      def raise_event(event_name, data = nil)
```

Comment:
I would avoid the word "raise" because it can be confused with Ruby built-in error raising (like raising exceptions).

Please keep the name consistent with TK's terminology by renaming to `event_generate`. It's OK to shadow it if you are adding value on top of it (like by auto-serializing to YAML). That way, this becomes the "smart" version of `tk.event_generate`.